sway/commands/shortcuts_inhibitor.c

```diff
@@ -0,0 +1,40 @@
+#include "sway/commands.h"
+#include "sway/input/seat.h"
+#include "sway/input/input-manager.h"
+
+struct cmd_results *cmd_shortcuts_inhibitor(int argc, char **argv) {
+	struct cmd_results *error = checkarg(argc, "shortcuts_inhibitor",
+			EXPECTED_EQUAL_TO, 1);
+	if (error) {
+		return error;
+	}
+
+	struct sway_seat *seat = config->handler_context.seat;
```

Comment:
Done. While at it I took inspiration from `pointer_constraint` and added a per-`seat_config` (and thus with `*` potentially global) enable/disable default. This actually allows the user to decide when to activate an inhibitor.

The inhibitor actions don't seem to fit into this all that well. I had to redefine the wildcard `seat *` to mean "current seat"  to keep semantics. What do you think of this?

What's also left is the question how the user could get some indication of an existing inhibitor on a window and its state. Maybe a border color change? I had thought of a criteria for that. What do you think and how would I go about it?

Reply:
I have the criteria coded up but it doesn't fully work because criteria are only executed once. So a config like:
```
for_window [shortcuts_inhibitor=active] title_format !+%title
for_window [shortcuts_inhibitor=inactive] title_format !-%title
for_window [shortcuts_inhibitor=absent] title_format %title
```
will happily upgrade the window title to '!+%title` as an inhibitor becomes active but not back down if it goes inactive or gets destroyed.

Now I'm thinking of executing a configurable command as the inhibitor state changes but I can't find any example for such a `command_to_exec_on_some_condition` hook-like functionality. What would the sway-native approach be here? Have a script listen to sway-ipc and send the `title_format` command?

Reply:
> The inhibitor actions don't seem to fit into this all that well. I had to redefine the wildcard seat * to mean "current seat" to keep semantics. What do you think of this?

There is already `seat -` for the current seat. `seat *` should work fine for `enable|disable`. Either return a result with the status `CMD_INVALID` for `activate|deactivate` on the wildcard seat or activate/deactivate all shortcut inhibitors. I'm not really sure which is better.

> What's also left is the question how the user could get some indication of an existing inhibitor on a window and its state. Maybe a border color change? I had thought of a criteria for that. What do you think and how would I go about it?
> [..]
> What would the sway-native approach be here? Have a script listen to sway-ipc and send the title_format command?

Information for none of the inhibitor types is currently exposed via IPC. It could be added to get_tree response though. Any other change would be out of scope for this PR and would have to be discussed as a separate one.

Reply:
Done. I had totally missed `seat -`. I've gone the route of applying the active/inactive/toggle actions to the current/named/all seat(s) because it seems most consistent to me.

Is there a way to avoid iterating all seats and all inhibitors when trying to add an inhibitor indicator to `get_tree`? For now I've added a note to the manpage that indicator indication is up to the client.